### PR TITLE
Add buildings

### LINKED ISSUE #7: Create buildings models

### PR DESCRIPTION
**Pull request by MendelDamian**

### CODE REVIEW

File: game/buildings.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+@dataclass
+class Barracks(Building):
+    MAX_LEVEL = 3
+    upgrade_time: timedelta = timedelta(minutes=5)
+
+    # Costs
+    wood_cost: int = 200
+    clay_cost: int = 170
+    iron_cost: int = 90
```
Comment (MendelDamian) [inline game/buildings.py]: Cost values should be re-considered

File: game/buildings.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+    def get_upgrade_cost(self) -> dict:
+        return {
+            "wood": self.wood_cost * self.level,
+            "clay": self.clay_cost * self.level,
+            "iron": self.iron_cost * self.level,
+        }
+
+    def get_upgrade_time(self) -> timedelta:
+        return self.upgrade_time * self.level
```
Comment (MendelDamian) [inline game/buildings.py]: as well as this type of calculating cost / duration

File: game/buildings.py [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    # Resources per second
+    wood_production: float = 0
+    clay_production: float = 0
+    iron_production: float = 0
+
+
+@dataclass
+class TownHall(Building):
+    MAX_LEVEL = 3
+    upgrade_time: timedelta = timedelta(minutes=10)
```
Comment (Blindeen) [inline game/buildings.py]: We should consider variable upgrade time depending on building level. It's only additional feature and it's up to you whether you want to apply it or not

**Review by Blindeen — APPROVED** [review]
Well done. I am amazed how clean you managed to create buildings models 🧹

**Review by Blindeen — APPROVED** [review]
Looks quality 🪩

**Pull request merged**